src/middlewared/middlewared/plugins/jail.py

```diff
@@ -244,6 +247,7 @@ def do_create(self, data):
         `branch` is the FreeNAS repository branch to use as the base for the `plugin_repository`. The default is to
         use the current system version. Example: 11.3-RELEASE.
         """
+        data['plugin_repository'] = data.get('plugin_repository', self.default_repo())
```

Comment:
data.get('plugin_repository') or self.default_repo()

otherwise default_repo is evaluated even if plugin_repository is passed

Reply:
Is there a drawback to that ? If it is passed, the passed value will still be respected. Idea is if user passes empty string, we should raise an error

Reply:
The drawback is that we will be calling a function that will not be used